Add Chatbot Arena Web Server Documentation and Fix Latex Display (#3615)
lm-sys/FastChat: README.md

By following these steps, you will be able to serve your models using the web UI. You can open your browser and chat with a model now.
If the models do not show up, try to reboot the gradio web server.

> ## Launch Chatbot Arena

Reviewer: ```suggestion
## Launch Chatbot Arena (side-by-side battle UI)
```

Currently, Chatbot Arena is powered by FastChat. Here is how you can launch an instance of Chatbot Arena locally.

> Create a file `api_endpoint.json` and record the the api endpoints of the models you want to serve, for example:

Reviewer: ```suggestion
Create a file `api_endpoint.json` with the api endpoints of the models you want to serve, for example:
```

```
{
    "gpt-4o-2024-05-13": {
        "model_name": "gpt-4o-2024-05-13",
        "api_base": "https://api.openai.com/v1",
        "api_type": "openai",
        "api_key": [Insert API Key],
        "anony_only": false
    }
}
```

> For anthropic model, set `api_type` to `"anthropic_message"` and replace `api_key` with your anthropic key. For gemini model, set `api_type` to `"gemini"` and replace `api_key` with your gemini key. For additional information, you can refer to `fastchat/serve/api_provider.py` for implementation details of other model types.

Reviewer: can we copy the content from
https://github.com/lm-sys/FastChat/blob/main/docs/model_support.md#api-based-models

it's more complete

> Then make sure to export `OPENAI_API_KEY` with your openai key. For anthropic model, set `api_type` to `"anthropic_message"` and export `ANTHROPIC_API_KEY` with your anthropic key. For gemini model, set `api_type` to `"gemini"` and export `GEMINI_API_KEY` with your gemini key. For additional information, you can refer to `fastchat/serve/api_provider.py` for implementation details of other model types.

Reviewer: i dont think user needs to export `OPENAI_API_KEY` if you already specify api_key in the `api_endpoint.json`

Author: Oh right.

By following these steps, you will be able to serve your models using the web UI. You can open your browser and chat with a model now.
If the models do not show up, try to reboot the gradio web server.

## Launch Chatbot Arena (side-by-side battle UI)

> Currently, Chatbot Arena is powered by FastChat. Here is how you can launch an instance of Chatbot Arena locally.
> 
> To support an API-based model, consider using the existing OpenAI api code for example.
> If the model is compatible with OpenAI APIs, then only modifying the configuration file is required without any additional code.
> For custom protocols, implementation of a streaming generator in [fastchat/serve/api_provider.py](https://github.com/lm-sys/FastChat/blob/main/fastchat/serve/api_provider.py) is required, following the provided examples. Currently, FastChat is compatible with OpenAI, Anthropic, Google Vertex AI, Mistral, Nvidia NGC, YandexGPT and Reka. For more information about model support, please refer to the model support [Doc](./docs/model_support.md).

Reviewer: ```suggestion
Currently, Chatbot Arena is powered by FastChat. Here is how you can launch an instance of Chatbot Arena locally.

FastChat supports popular API-based models such as OpenAI, Anthropic, Gemini, Mistral and more. To add a custom API, please refer to the model support [doc](./docs/model_support.md) for more details. Below we take OpenAI models as an example.
```

Create a JSON configuration file `api_endpoint.json` with the api endpoints of the models you want to serve, for example:
```
{
    "gpt-4o-2024-05-13": {
        "model_name": "gpt-4o-2024-05-13",
        "api_base": "https://api.openai.com/v1",
        "api_type": "openai",
        "api_key": [Insert API Key],
        "anony_only": falseg
    }
}
```

> For anthropic model, set `api_type` to `"anthropic_message"` and replace `api_key` with your anthropic key. For gemini model, set `api_type` to `"gemini"` and replace `api_key` with your gemini key.

Reviewer: ```suggestion
For Anthropic models, specify `"api_type": "anthropic_message"` with your Anthropic key. Similarly, for gemini model, specify `"api_type": "gemini"`. More details can be found in [api_provider.py](https://github.com/lm-sys/FastChat/blob/main/fastchat/serve/api_provider.py).
```

> If you want to serve your own model using local gpus, following the instructions in [Serving with Web GUI](#serving-with-web-gui).

Reviewer: ```suggestion
To serve your own model using local gpus, follow the instructions in [Serving with Web GUI](#serving-with-web-gui).
```

> To launch a gradio web server, run `gradio_web_server_multi.py`.

Reviewer: ```suggestion
Now you're ready to launch the server:
```
